api/users.go

```diff
@@ -10,7 +10,7 @@ import (
 func initUserHandlers(userRouter fiber.Router) {
 	userRouter.Post("/", userCreateHandler())
 
-	userRouter.Use(auth.CheckToken())
+	userRouter.Use(auth.CheckToken(db.UserUT))
```

Comment:
Why use this as the default call? I guess Im not 100% sure on how access is checked here, if the function is called once with a default

Reply:
Not sure if I understand what exactly you mean but the way this works is just like the route handlers : CheckToken is a function that returns another function and said other function is the actual middleware that will be called when someone sends a request. And that actual middleware will compare the user from the jwt token to the type it was given on initalization

Reply:
Arent Admins then disallowed from using this route?

Reply:
If we were to only check against the given type yes but CheckToken() uses userTypeIncludes(). So AdminUT works for UserUT restricted routes but not vice versa. There might even be a prettier way to do this i.e. if user types were numbers we could simply do ">=" checks but I don't want to do that to the frontend team right now.